scripts/crawl/indexer.js

```diff
@@ -92,15 +188,26 @@ function main() {
         // not all the entries have keywords
         if (sitemapDatabase[url].text) {
             var keywords_string = sitemapDatabase[url].text.match(/<!--[\s]*?(keywords:)(.*?)[\s]*?-->/gi);
-            if(!keywords_string || keywords_string.length === 0){ return; }
-            
+            if (!keywords_string || keywords_string.length === 0) {
+                return;
+            }
+
             keywords_string = keywords_string[0].replace(/<!--[\s]*?(keywords:)(.*?)[\s]*?-->/gi, '$2').trim();
             var keywords_array = keywords_string.split(' ');
 
             keywords_array.forEach((keyword) => {
                 if (!BLACK_LIST.has(keyword)) {
-                    // set the keyword index to 1 (only count a keyword as 1)
-                    keywordIndex[keyword] = { [url]: 1 };
+                    if (keywordIndex[keyword]) {
+                        // this keyword has been indexed at least once before
+                        if (keywordIndex[keyword][url]) {
+                            keywordIndex[keyword][url] += 1;
+                        } else {
+                            keywordIndex[keyword][url] = 1;
+                        }
+                    } else {
+                        // we never see this keyword before
+                        keywordIndex[keyword] = { [url]: 1 };
+                    }
```

Comment:
are we anticipating adding multiples of the same keyword on a page?
```
<!-- KEYWORDS: keyword1 keyword1 keyword1 keyword2 keyword3 -->
```

Reply:
Ah, I see what happened here. We definitely need the outer else, but I don't think we need the inner if/else...i think we can just set it to one, assuming we won't have duplicate keywords on the same page (I don't think we want to since that might cause some weirdness with the weighting algorithm).

Reply:
Yeah, I was thinking of the inner else as a try-catch thing... wait, this is not typescript :P I can take that logic out.